src/content/developers/docs/scaling/zk-rollups/index.md

```diff
@@ -1,35 +1,221 @@
+
+3. **L2 operator fees**: This is the amount paid to the rollup operator as compensation for computational costs incurred in processing transactions, much like miner fees on Ethereum.
+
+4. **Proof generation and verification**: ZK-rollup operators must produce validity proofs for transaction batches, which is resource-intensive. Verifying zero-knowledge proofs on Mainnet also costs gas (~ 500,000 gas).
+
+Apart from batching transactions, ZK-rollups reduce fees for users by compressing transaction data. You can see [here](https://l2fees.info/) for a real-time overview of how it costs to use Ethereum ZK-rollups.
+
+## How do ZK-rollups scale Ethereum? {#scaling-ethereum-with-zk-rollups}
+
+### Transaction data compression {#transaction-data-compression}
+
+ZK-rollups extend the throughput on Ethereum’s base layer by taking computation off-chain, but the real boost for scaling comes from compressing transaction data. Ethereum’s [block size](/developers/docs/blocks/#block-size) limits the data each block can hold and, by extension, the number of transactions processed per block. By compressing transaction-related data, ZK-rollups significantly increase the number of transactions processed per block.
+
+ZK-rollups can compress transaction data better than optimistic rollups since they don't have to post all the data required to validate each transaction. They only have to post the minimal data required to rebuild the latest state of accounts and balances on the rollup.
+
+### Recursive proofs {#recursive-proofs}
+
+An advantage of zero-knowledge proofs is that proofs can verify other proofs. For example, a single ZK-SNARK can verify other ZK-SNARKs. Such "proof-of-proofs" are called recursive proofs and dramatically increase throughput on ZK-rollups.
+
+Currently, validity proofs are generated on a block-by-block basis and submitted to the L1 contract for verification. However, verifying single block proofs limits the throughput that ZK-rollups can achieve since only one block can be finalized when the operator submits a proof.
+
+Recursive proofs, however, make it possible to finalize several blocks with one validity proof. This is because the proving circuit recursively aggregates multiple block proofs until one final proof is created. The L2 operator submits this recursive proof, and if the contract accepts it, all the relevant blocks will be finalized instantly. With recursive proofs, the number of ZK-rollup transactions that can be finalized on Ethereum at intervals increases.
+
+### Pros and cons of zk-rollups {#zk-rollups-pros-and-cons}
+
+| Pros                                                                                                                                                                                                  | Cons                                                                                                                                                                                               |
```

Comment:
The table seems to be rendering in a way where the pros column is much narrower than the cons. Not a blocker, but if we know a way to sort this out would be nice.

Reply:
Hm. I'll check the doc and see if I can fix it.